assets/js/googlesitekit/data/create-fetch-store.test.js

```diff
@@ -0,0 +1,319 @@
+/**
+ * API function to create fetch store tests.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+import isPlainObject from 'lodash/isPlainObject';
+
+/**
+ * WordPress dependencies
+ */
+import { createRegistry } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import {
+	muteConsole,
+	subscribeUntil,
+	unsubscribeFromAll,
+} from 'tests/js/utils';
+import { createFetchStore } from './create-fetch-store';
+
+const STORE_NAME = 'test/some-data';
+const STORE_PARAMS = {
+	baseName: 'getSomeData',
+	controlCallback: ( params ) => {
+		const { aParam, objParam } = params;
+		return API.get( 'core', 'test', 'some-data', {
+			aParam,
+			objParam,
+		} );
+	},
+	reducerCallback: ( state, response, params ) => {
+		const { aParam } = params;
+		return {
+			...state,
+			data: {
+				...( state.data || {} ),
+				[ aParam ]: response,
+			},
+		};
+	},
+	argsToParams: ( objParam, aParam ) => {
+		invariant( isPlainObject( objParam ), 'objParam is required.' );
```

Comment:
I wonder if having checks like this in `argsToParams` is the right approach; see my comment above about catching errors in this function.